Gordon360/Services/ProfileService.cs

```diff
@@ -301,6 +306,26 @@ public async Task UpdateMobilePrivacyAsync(string username, string value)
             _context.SaveChanges();
         }
 
+        /// <summary>
+        /// privacy setting of mobile phone.
+        /// </summary>
+        /// <param name="username">AD Username</param>
+        /// <param name="value">Y or N</param>
+        /// 
+        /// not being used for now
+        public async Task UpdatePlannedGraduationYearPrivacyAsync(string username, string value)
+        {
+            var account = _accountService.GetAccountByUsername(username);
+            // Update value in cached data
+            var student = _context.CUSTOM_PROFILE.FirstOrDefault(x => x.username == username);
+            if (student == null)
+            {
+                await _context.CUSTOM_PROFILE.AddAsync(new CUSTOM_PROFILE { username = username, IsPlannedGradYearPrivate = (value == "Y" ? 1 : 0) });
+            }
+            student.IsPlannedGradYearPrivate = (value == "Y" ? 1 : 0);
```

Comment:
```suggestion
            student.IsPlannedGradYearPrivate = (value == "Y" ? 1 : 0);
```
Why is this privacy variable switch between so many types? It should just be consistent to bool

Reply:
In the UI we are doing the same as for other privacy changing routes:
![image](https://github.com/gordon-cs/gordon-360-api/assets/115328139/08b3be28-2c2a-48a6-8e76-99fc23d3987d)
and it is stored as 
![image](https://github.com/gordon-cs/gordon-360-api/assets/115328139/6728510f-aeba-4e56-bde4-e56b143e0b90)

Reply:
right which is why I think it should be switched to `bit` which is translated to boolean

Reply:
I tried to make it more consistent now

Reply:
@amos-cha now, instead of converting from boolean to string and then to int, it is a bool until the very end where is converted to string to store in the database.

Reply:
Is this good enough for now? this route is actually not used yet in the frontend because the place where it will be stored the information will be changed to another table.

Reply:
I think this is a conversation we need to have offline because there are things I don't really understand about this.

Reply:
The table changed was CUSTOM_PROFILE from CCT . I tried clicking on the EF Core Power Tools - Refresh, but nothing changed.  
![image](https://github.com/gordon-cs/gordon-360-api/assets/115328139/2c62881b-3f03-43af-a50a-6189ac9b2ec9)